modules/how-to-plan-your-environment-according-to-application-requirements.adoc

```diff
@@ -69,3 +69,99 @@ do not. Most Java applications and applications that use huge pages are examples
 of applications that would not allow for overcommitment. That memory can not be
 used for other applications. In the example above, the environment would be
 roughly 30 percent overcommitted, a common ratio.
+
+The application Pods can access a service either by using environment variables or DNS. 
+If using environment variables, for each active service the variables are injected by the 
+kubelet when a Pod is run on a node. A cluster-aware DNS server watches the Kubernetes API 
+for new services and creates a set of DNS records for each one. If DNS is enabled throughout 
+your cluster, then all Pods should automatically be able to resolve services by their DNS name. 
+Service discovery using DNS can be used in case you must go beyond 5000 services. When using 
+environment variables for service discovery, the argument list exceeds the allowed length after 
+5000 services in a namespace, then the Pods and deployments will start failing. Disable the service 
+links in the deployment's service specification file to overcome this:
+```
```

Comment:
Sorry, we also need a [source, yaml] just before the code block per https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#code-blocks. The code block must also start and end in four dashes:

    ----

Reply:
Fixed it, thanks.